tegra-fuse.c

```diff
@@ -170,10 +188,13 @@ tegra_fuse_context_open (const char *basepath)
 	ssize_t typelen;
 	tegra_soctype_t soc;
 	int fd;
-	unsigned long chipid;
+	unsigned short chipid;
 	char soctype[65];
-
-	fd = open("/sys/module/tegra_fuse/parameters/tegra_chip_id", O_RDONLY);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+	fd = open("/sys/module/fuse_burn/parameters/tegra_chip_uid", O_RDONLY);
+#else
+        fd = open("/sys/module/tegra_fuse/parameters/tegra_chip_id", O_RDONLY);
+#endif
```

Comment:
Just try opening one, and if that fails, try opening the other.  You'll need to keep track of which one, and use the right logic based on that, rather than using these kernel version-based ifdefs.